controllers/update_pods_test.go

```diff
@@ -246,5 +249,98 @@ var _ = Describe("update_pods", func() {
 				Expect(updates).To(HaveLen(1))
 			})
 		})
+
+		When("max unavailable pods is set with a percent value and there are process groups with pods in pending status", func() {
+			BeforeEach(func() {
+				expectedError = true
+				cluster.Spec.MaxUnavailablePods = intstr.FromString("1%")
+				Expect(k8sClient.Update(context.TODO(), cluster)).NotTo(HaveOccurred())
+
+				var numPendingPods int
+				for _, processGroup := range cluster.Status.ProcessGroups {
+					if processGroup.ProcessClass.IsStateful() {
+						processGroup.ProcessGroupConditions = append(processGroup.ProcessGroupConditions, fdbv1beta2.NewProcessGroupCondition(fdbv1beta2.PodPending))
+						numPendingPods++
+						if numPendingPods == 8 {
+							break
+						}
+					}
+				}
+			})
+
+			It("should return an error and nil updates", func() {
+				Expect(updates).To(BeNil())
+				Expect(expectedError).To(BeTrue())
+				Expect(err.Error()).Should(ContainSubstring("cluster has 8 Pods that are pending or missing and 0 Pods will be updated, exceeding cluster.Spec.MaxUnavailablePods: 1"))
+			})
+		})
+
+		When("max unavailable pods is set with an int value and there are two process groups with pods in pending status", func() {
+			BeforeEach(func() {
+				expectedError = true
+				cluster.Spec.MaxUnavailablePods = intstr.FromInt(1)
+				Expect(k8sClient.Update(context.TODO(), cluster)).NotTo(HaveOccurred())
+
+				var numPendingPods int
+				for _, processGroup := range cluster.Status.ProcessGroups {
+					if processGroup.ProcessClass.IsStateful() {
+						processGroup.ProcessGroupConditions = append(processGroup.ProcessGroupConditions, fdbv1beta2.NewProcessGroupCondition(fdbv1beta2.PodPending))
+						numPendingPods++
+						if numPendingPods == 2 {
+							break
+						}
+					}
+				}
+			})
+
+			It("should return an error and nil updates", func() {
+				Expect(updates).To(BeNil())
+				Expect(expectedError).To(BeTrue())
+				Expect(err.Error()).Should(ContainSubstring("cluster has 2 Pods that are pending or missing and 0 Pods will be updated, exceeding cluster.Spec.MaxUnavailablePods: 1"))
+			})
+		})
+
+		When("max unavailable pods is greater than the number of pods requiring update plus the number of missing or pending pods", func() {
+			BeforeEach(func() {
+				expectedError = false
+				// Set max unavailable pods to 100 (arbitrary large number)
+				cluster.Spec.MaxUnavailablePods = intstr.FromInt(100)
```

Comment:
Can we use a more reasonable value here to e.g. allow only to update 1 Pod. Basically the value 100 in this test case is the same as setting the limit to 0 (not having any limit) as the Pods created by this cluster is much smaller than 100.